File: 프로그래머스/lv2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        
        for(int i = 1; n-((i*(i-1))/2) > 0; i++){
            if((n-((i*(i-1))/2))% i == 0){
                answer++;
            }
        }
        return answer;
    }

    public static void main(String[] args) {
        Solution s = new Solution();
        int n = 15;
        int answer = s.solution(n);
        System.out.println("answer = " + answer);
    }
}

File: JavaScript/프로그래머스/2/12924. 숫자의 표현/숫자의 표현.js
function solution(n) {
    let answer = 0;

    for (let m = 1; m * (m + 1) / 2 <= n; m++) {
        if ((n - (m * (m - 1)) / 2) % m === 0) {
            answer++;
        }
    }

    return answer;
}